src/setup_payload/AdditionalDataPayloadGenerator.cpp

```diff
@@ -51,35 +51,34 @@ AdditionalDataPayloadGenerator::generateAdditionalDataPayload(uint16_t lifetimeC
 {
     System::PacketBufferTLVWriter writer;
     TLVWriter innerWriter;
-    char rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
+    uint8_t rotatingDeviceIdBuffer[RotatingDeviceId::kHexMaxLength];
     size_t rotatingDeviceIdBufferSize = 0;
 
     // Initialize TLVWriter
     writer.Init(chip::System::PacketBufferHandle::New(chip::System::PacketBuffer::kMaxSize));
 
-    ReturnErrorOnFailure(writer.OpenContainer(AnonymousTag, kTLVType_Structure, innerWriter));
+    ReturnErrorOnFailure(writer.OpenContainer(RotatingDeviceId::DiscoveryExtensionDescriptorTag, kTLVType_Structure, innerWriter));
```

Comment:
```suggestion
    ReturnErrorOnFailure(writer.OpenContainer(AnonymousTag, kTLVType_Structure, innerWriter));
```
Yes, this code was already correct.